### PR TITLE
[1.26.1] Automated cherry pick of #113933: Limit request retrying to []byte request bodies

### PR DESCRIPTION
**Pull request by liggitt**
Cherry pick of #113933 on release-1.26.

#113933: Limit request retrying to []byte request bodies

For details on the cherry pick process, see the [cherry pick requests](https://git.k8s.io/community/contributors/devel/sig-release/cherry-picks.md) page.

```release-note

```

### CODE REVIEW

**Review by MadhavJivrajani — APPROVED** [review]
/lgtm

**Comment by saschagrunert** [comment]
Will merge after v1.26.0, means we can:

/milestone clear

**Comment by liggitt** [comment]
PRs against release branches get automatically milestoned, and should probably keep that milestone (at least once merged) to be able to find the PRs that went into that release, right?

**Comment by saschagrunert** [comment]
> PRs against release branches get automatically milestoned, and should probably keep that milestone (at least once merged) to be able to find the PRs that went into that release, right?

I thought we only milestone PRs for the latest minor (v1.26) when they should merge into the `master` branch, because then they get synched to the actual release branch. We can re-add the milestone, but it will have no effect to later cherry-pick approvals or something. My main intention was to double check that everything in `master` is also in `release-1.26` because the synch only happens every 6 hours. Means clearing the milestone before the release should be a good indicator that nothing is in the queue, even is we target non `master`.

**Comment by liggitt** [comment]
>  We can re-add the milestone, but it will have no effect to later cherry-pick approvals or something.

Yeah, I wasn't expecting it to. The milestone gets added by the bot like this: https://github.com/kubernetes/kubernetes/pull/113988#event-7839690073

**Comment by saschagrunert** [comment]
> > We can re-add the milestone, but it will have no effect to later cherry-pick approvals or something.
> 
> Yeah, I wasn't expecting it to. The milestone gets added by the bot like this: [#113988 (comment)](https://github.com/kubernetes/kubernetes/pull/113988#event-7839690073)

Do you think it makes sense to exclude applying milestones to release branches which have no final minor?

It would make things inconsistent somehow but could help us maintaining the latest milestone.

**Comment by liggitt** [comment]
> Do you think it makes sense to exclude applying milestones to release branches which have no v1.x.0 tag?

I think it's better to be consistent that all PRs merged to or targeting a minor version get the milestone applied.

**Comment by saschagrunert** [comment]
Makes sense
/milestone v1.26

**Comment by xmudrii** [comment]
@liggitt I'm waiting for #113993 to pass and then I'll approve all four cherry-picks.

**Review by xmudrii — APPROVED** [review]
For RelEng:
/lgtm
/approve

**Pull request merged**
